scheduler/scheduler-server/src/main/java/org/ow2/proactive/scheduler/authentication/ManageUsers.java

```diff
@@ -0,0 +1,579 @@
+    public static final String GROUPFILE_OPTION_NAME = "groupfile";
+
+
+    /**
+     * Entry point
+     *
+     * @param args arguments, try '-h' for help
+     * @throws IOException
+     * @throws ParseException
+     * @see org.ow2.proactive.authentication.crypto.Credentials
+     */
+    public static void main(String[] args) {
+
+        try {
+
+            manageUsers(args);
+
+        } catch (ManageUsersException e) {
+            System.err.println("ERROR : " + e.getMessage());
+            if (e.getCause() != null) {
+                System.err.println(e.getCause().getMessage());
+            }
+            if (e.getAdditionalInfo() != null) {
+                System.out.println(e.getAdditionalInfo());
+            }
+            System.exit(1);
+        }
+
+        System.exit(0);
+    }
+
+    public static void manageUsers(String[] args) throws ManageUsersException {
+        SecurityManagerConfigurator.configureSecurityManager(CreateCredentials.class.getResource(
+                "/all-permissions.security.policy").toString());
+
+        Console console = System.console();
+        /**
+         * default values
+         */
+        boolean nonInteractive = true;
+        String pubKeyPath = null;
+        PublicKey pubKey = null;
+        UserInfo userInfo = new UserInfo();
+        String loginFilePath = getLoginFilePath();
+        String groupFilePath = getGroupFilePath();
+        Action action = null;
+
+        Options options = new Options();
+
+        CommandLine cmd = getCommandLine(args, loginFilePath, groupFilePath, options);
+
+        if (cmd.hasOption(HELP_OPTION_NAME)) {
+            displayHelp(options);
+            return;
+        }
+
+        action = Action.getAction(cmd);
+
+        if (cmd.hasOption(LOGIN_OPTION_NAME)) {
+            userInfo.setLogin(cmd.getOptionValue(LOGIN_OPTION_NAME));
+        }
+        if (cmd.hasOption(PWD_OPTION_NAME)) {
+            userInfo.setPassword(cmd.getOptionValue(PWD_OPTION_NAME));
+        }
+        if (cmd.hasOption(GROUPS_OPTION_NAME)) {
+            String groupString = cmd.getOptionValue(GROUPS_OPTION_NAME);
+            userInfo.setGroups(groupString.split(","));
+        }
+
+        if (cmd.hasOption(LOGINFILE_OPTION_NAME)) {
+            loginFilePath = cmd.getOptionValue(LOGINFILE_OPTION_NAME);
+        }
+        if (cmd.hasOption(GROUPFILE_OPTION_NAME)) {
+            groupFilePath = cmd.getOptionValue(GROUPFILE_OPTION_NAME);
+        }
+        if (cmd.hasOption(KEYFILE_OPTION_NAME)) {
+            pubKeyPath = cmd.getOptionValue(KEYFILE_OPTION_NAME);
+        }
+
+        if (pubKeyPath == null) {
+            pubKeyPath = getPublicKeyFilePath();
+        }
+        try {
+            pubKey = Credentials.getPublicKey(pubKeyPath);
+        } catch (KeyException e) {
+            exitWithErrorMessage("Could not retrieve public key from '" + pubKeyPath, null, e);
+        }
+
+        switch (action) {
```

Comment:
![MAJOR](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-major.png) Add a default case to this switch. [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](http://master.ci.activeeon.com:9000/coding_rules#rule_key=squid%3ASwitchLastCaseIsDefaultCheck)